src/components/Header.tsx

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useContext, useState } from 'react';
+import { DispatchContext, StateContext } from './TodosContext';
+import { createPost, updateTodo } from '../api/todos';
+
+export const Header = () => {
+  const { todos } = useContext(StateContext);
+  const [title, setTitle] = useState('');
+  const dispatch = useContext(DispatchContext);
+  const USER_ID = 56;
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const notCompletedTodos = todos.filter(todo => !todo.completed);
+  const isAllCompleted = todos.every(todo => todo.completed);
+  const toggleAllTodos = isAllCompleted ? completedTodos : notCompletedTodos;
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+  };
+
+  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title.trim()) {
+      dispatch({
+        type: 'errorMessage',
+        payload: 'Title should not be empty',
+      });
+
+      return;
+    }
+
+    const newTodo = {
+      title,
+      completed: false,
+      userId: USER_ID,
+      id: +Date.now(),
```

Comment:
Try to use crypto.randomUUID() for id 

https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID